File: backend/Models/Workspace.cs
using System.ComponentModel.DataAnnotations;

namespace backend.Models;

public class Workspace
{
    [Key]
    public Guid Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? CompanyName { get; set; }
    public string? CompanyUrl { get; set; }
    public string? PhoneNumber { get; set; }
    public string Currency { get; set; } = "USD";
    public decimal DefaultTaxRate { get; set; }
    public string DefaultPaymentTerms { get; set; } = "uponReceipt";
    public string? TaxRegistrationNumber { get; set; }
    public string? AddressLine1 { get; set; }
    public string? AddressLine2 { get; set; }
    public string? City { get; set; }
    public string? State { get; set; }
    public string? PostalCode { get; set; }
    public string? Country { get; set; }
    public CompanySize Size { get; set; } = CompanySize.Solo;
    public Guid CreatedByUserId { get; set; }
    public User? CreatedByUser { get; set; }
    public string? LogoUrl { get; set; }
    public string Theme { get; set; } = "light";
    public string Category { get; set; } = string.Empty;
    public ICollection<User> Users { get; set; } = [];
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}

public enum CompanySize
{
    Solo,         // 0-1 employees
    Small,        // 2-5
    Medium,       // 6-10
    Large,        // 10+
}

File: frontend/src/types/Workspace.ts
import { CompanySize } from '../constants/Enumeration/WorkspaceEnum/WorkspaceEnum';
import { TUser, TUserLookup } from './User';

export type TWorkspace = {
  id: string;
  name: string;
  companyName?: string;
  companyUrl?: string;
  phoneNumber?: string;
  currency: string;
  defaultTaxRate: number;
  defaultPaymentTerms: string;
  taxRegistrationNumber?: string;
  addressLine1?: string;
  addressLine2?: string;
  city?: string;
  state?: string;
  postalCode?: string;
  country?: string;
  size: CompanySize;
  createdByUserId: string;
  createdByUser?: TUser;
  logoUrl?: string;
  theme: string;
  category: string;
  users: TUser[];
  createdAt: string;
  updatedAt: string;
};

export type TGetWorkspace = {
  id: string;
  name: string;
  companyName?: string;
  companyUrl?: string;
  phoneNumber?: string;
  currency: string;
  defaultTaxRate: number;
  defaultPaymentTerms: string;
  taxRegistrationNumber?: string;
  addressLine1?: string;
  addressLine2?: string;
  city?: string;
  state?: string;
  postalCode?: string;
  country?: string;
  size: CompanySize;
  createdByUser?: TUserLookup;
  logoUrl?: string;
  theme: string;
  category: string;
  users: TUserLookup[];
  createdAt: string;
  updatedAt: string;
};

export type TAddWorkspace = {
  name: string;
  companyName?: string;
  companyUrl?: string;
  phoneNumber?: string;
  currency: string;
  defaultTaxRate: number;
  defaultPaymentTerms: string;
  taxRegistrationNumber?: string;
  addressLine1?: string;
  addressLine2?: string;
  city?: string;
  state?: string;
  postalCode?: string;
  country?: string;
  size: CompanySize;
  createdByUserId: string;
  logoUrl?: string;
  theme: string;
  category: string;
};

export type TUpdateWorkspace = {
  id: string;
  name?: string;
  companyName?: string;
  companyUrl?: string;
  phoneNumber?: string;
  currency?: string;
  defaultTaxRate?: number;
  defaultPaymentTerms?: string;
  taxRegistrationNumber?: string;
  addressLine1?: string;
  addressLine2?: string;
  city?: string;
  state?: string;
  postalCode?: string;
  country?: string;
  size?: CompanySize;
  logoUrl?: string;
  theme?: string;
  category?: string;
};

export type TWorkspaceLookup = {
  id: string;
  name: string;
};
